code/game/objects/effects/effect_system/effects_sparks.dm

```diff
@@ -26,24 +26,50 @@
 	return INITIALIZE_HINT_LATELOAD
 
 /obj/effect/particle_effect/sparks/LateInitialize()
+	RegisterSignals(src, list(COMSIG_MOVABLE_CROSS, COMSIG_MOVABLE_CROSS_OVER), PROC_REF(sparks_touched))
 	flick(icon_state, src)
 	playsound(src, SFX_SPARKS, 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
 	var/turf/T = loc
 	if(isturf(T))
-		T.hotspot_expose(1000,100)
+		affect_location(T, TRUE) // just_initialized set to TRUE
 	QDEL_IN(src, 20)
 
 /obj/effect/particle_effect/sparks/Destroy()
 	var/turf/T = loc
 	if(isturf(T))
-		T.hotspot_expose(1000,100)
+		affect_location(T)
+	UnregisterSignal(src, list(COMSIG_MOVABLE_CROSS, COMSIG_MOVABLE_CROSS_OVER))
```

Comment:
You don't need to UnregisterSignal on src

Reply:
You resolved this but didn't do anything about it

Reply:
Still not fixed

Reply:
my man please

Reply:
on god i have pushed it twice now!!!!!

Reply:
alright there we go.

Reply:
It's still here and on your remote
<https://github.com/Metekillot/tgstation/blob/23ca8aaed9fd3b8d6e2a3b07adc7b9fbea76a420/code/game/objects/effects/effect_system/effects_sparks.dm#L41>

What are you actually doing to verify that it's gone before you keep resolving this comment?
Are you double sure that you aren't just fixing the wrong line in a file and not the one we want you to delete?

Reply:
my eyes were glazing over that because of the problems with the renamed variables i had earlier. i removed it now.